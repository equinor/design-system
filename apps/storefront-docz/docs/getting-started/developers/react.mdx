---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
