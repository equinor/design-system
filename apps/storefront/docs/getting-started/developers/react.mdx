---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
