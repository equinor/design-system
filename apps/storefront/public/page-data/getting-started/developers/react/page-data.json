{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}