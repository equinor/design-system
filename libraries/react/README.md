@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
