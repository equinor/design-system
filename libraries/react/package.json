{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
