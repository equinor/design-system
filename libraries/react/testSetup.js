import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
