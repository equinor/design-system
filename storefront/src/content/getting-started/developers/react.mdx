---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
