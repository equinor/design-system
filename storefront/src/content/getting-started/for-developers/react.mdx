---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
