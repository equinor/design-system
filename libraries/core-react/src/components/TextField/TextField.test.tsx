/* eslint-disable no-undef */
import * as React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import 'jest-styled-components'
import styled from 'styled-components'
import { TextField } from '.'

afterEach(cleanup)

describe('TextField', () => {
  it('Has correct label text', () => {
    const labelText = 'Some label'
    const { queryByText } = render(
      <TextField id="test-label" label={labelText} />,
    )

    expect(queryByText(labelText)).toBeInTheDocument()
  })

  it('Has correct default value', () => {
    const value = 'Some value'
    const { queryByDisplayValue } = render(
      <TextField id="test-value" value={value} readOnly />,
    )
    const inputElement = queryByDisplayValue(value) as HTMLInputElement

    expect(inputElement.value).toBe(value)
  })

  it('Has correct updated value when changed', () => {
    let value = 'Initial value'
    const newValue = 'Updated value'
    const onChangeHandler = (ele) => {
      // eslint-disable-next-line
      value = ele.target.value
    }

    const { getByDisplayValue } = render(
      <TextField
        id="test-value-updated"
        value={value}
        onChange={onChangeHandler}
      />,
    )

    fireEvent.change(getByDisplayValue(value), {
      target: { value: newValue },
    })

    expect(value).toBe(newValue)
  })

  const StyledTextField = styled(TextField)`
    margin-top: 48px;
  `
  it('Can extend the css of the component', () => {
    const { container } = render(
      <StyledTextField id="test-css-extend" variant="error" />,
    )

    expect(container.firstChild).toHaveStyleRule('margin-top', '48px')
  })
})
