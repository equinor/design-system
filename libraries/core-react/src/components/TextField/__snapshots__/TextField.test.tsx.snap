// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Matches snapshot 1`] = `[Function]`;
